File: src/main/java/me/lukiiy/discordBridge/util/BotHelper.java
package me.lukiiy.discordBridge.util;

import me.lukiiy.discordBridge.DCBridge;
import net.dv8tion.jda.api.OnlineStatus;
import net.dv8tion.jda.api.entities.Activity;
import org.jetbrains.annotations.NotNull;

public class BotHelper {
    public static OnlineStatus getStatus(@NotNull String info) {
        try {return OnlineStatus.valueOf(info);}
        catch (IllegalArgumentException e) {
            DCBridge.log("\"" + info + "\" is not a valid status. Please change it in your config.yml.");
            return OnlineStatus.ONLINE;
        }
    }

    public static Activity getActivity(@NotNull String string) {
        String[] parts = string.split("\\s+", 2);
        String detail = parts.length > 1 ? parts[1] : string;

        try {return Activity.of(Activity.ActivityType.valueOf(parts[0].toUpperCase()), detail);}
        catch (IllegalArgumentException e) {
            DCBridge.log("\"" + string + "\" is not a valid activity type. Please change it in your config.yml.");
            return Activity.of(Activity.ActivityType.PLAYING, detail);
        }
    }
}

File: common/src/main/kotlin/me/lukiiy/discordBridge/utils/BotHelper.kt
package me.lukiiy.discordBridge.utils

import net.dv8tion.jda.api.OnlineStatus
import net.dv8tion.jda.api.entities.Activity

/**
 * Utility functions related to the bot
 */
object BotHelper {
    /**
     * Try to retrieve a valid status from a config string.
     *
     * @param string Something like "ONLINE"
     *
     * @return A valid [OnlineStatus] or an exception.
     */
    @JvmStatic
    @Throws(Exception::class)
    fun getStatus(string: String): OnlineStatus {
        try {
            return OnlineStatus.valueOf(string.uppercase())
        } catch (e: IllegalArgumentException) {
            throw Exception("\"$string\" cannot generate a valid status.", e)
        }
    }

    /**
     * Try to get a valid [Activity] from a formatted string.
     *
     * @param string Something like "playing Minecraft"
     *
     * @return A valid [Activity] or an exception.
     */
    @JvmStatic
    @Throws(Exception::class)
    fun getActivity(string: String): Activity {
        val parts = string.split("\\s+".toRegex(), 2)
        val detail = if (parts.size > 1) parts[1] else string

        try {
            return Activity.of(Activity.ActivityType.valueOf(parts[0].uppercase()), detail)
        } catch (e: IllegalArgumentException) {
            throw Exception("\"$string\" cannot generate a valid activity.", e)
        }
    }
}